otx/hpo/hpo_base.py

```diff
@@ -0,0 +1,305 @@
+        metric (str): Which score metric to use.
+        expected_time_ratio (Optional[Union[int, float]]): Time to use for HPO.
+                                            If HPO is configured automatically,
+                                            HPO use time about exepected_time_ratio *
+                                            train time after HPO times.
+        maximum_resource (Optional[Union[int, float]]): Maximum resource to use for training each trial.
+        subset_ratio (Optional[Union[float, int]]): ratio to how many train dataset to use for each trial.
+                                     The lower value is, the faster the speed is.
+                                     But If it's too low, HPO can be unstable.
+        min_subset_size (int) : Minimum size of subset. Default value is 500.
+        resume (bool): resume flag decide to use previous HPO results.
+                       If HPO completed, you can just use optimized hyper parameters.
+                       If HPO stopped in middle, you can resume in middle.
+        prior_hyper_parameters (Optional[Union[Dict, List[Dict]]]) = Hyper parameters to try first.
+        acceptable_additional_time_ratio (Union[float, int]) = Decide how much additional time can be acceptable.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    def __init__(
+        self,
+        search_space: Dict[str, Dict[str, Any]],
+        save_path: Optional[str] = None,
+        mode: str = "max",
+        num_trials: Optional[int] = None,
+        num_workers: int = 1,
+        num_full_iterations: Union[int, float] = 1,
+        non_pure_train_ratio: float = 0.2,
+        full_dataset_size: int = 0,
+        metric: str = "mAP",
+        expected_time_ratio: Optional[Union[int, float]] = None,
+        maximum_resource: Optional[Union[int, float]] = None,
+        subset_ratio: Optional[Union[float, int]] = None,
+        min_subset_size: int = 500,
+        batch_size_name: Optional[str] = None,
+        resume: bool = False,
+        prior_hyper_parameters: Optional[Union[Dict, List[Dict]]] = None,
+        acceptable_additional_time_ratio: Union[float, int] = 1.0,
+    ):
+        # pylint: disable=too-many-arguments, too-many-locals
+        check_mode_input(mode)
+        check_positive(full_dataset_size, "full_dataset_size")
+        check_positive(num_full_iterations, "num_full_iterations")
+        if not 0 < non_pure_train_ratio <= 1:
+            raise ValueError("non_pure_train_ratio should be between 0 and 1." f" Your value is {non_pure_train_ratio}")
+        if maximum_resource is not None:
+            check_positive(maximum_resource, "maximum_resource")
+        if num_trials is not None:
+            check_positive(num_trials, "num_trials")
+        check_positive(num_workers, "num_workers")
+        if subset_ratio is not None:
+            if not 0 < subset_ratio <= 1.0:
+                raise ValueError(
```

Comment:
Error message is different with `non_pure_train_ratio`, could you unify these?

Reply:
I'll update it.